collections/_article/godot-xr-progress-update-jan-2023.md

```diff
@@ -0,0 +1,79 @@
+
+### Godot documentation sprint
+
+Godot organised a documentation sprint this January to ensure the documentation for Godot 4 was brought up to spec in anticipation of Godot 4s stable release. XR of course was part of this sprint. Various documentation pages were completed and are now under review. Others are still being worked on. Expect to see the pages added to the documentation soon.
+
+### New backends in Godot 4
+
+#### OpenGL XR support in Godot 4
+
+For low end devices with limited Vulkan support, OpenGL was added as a backend to Godot 4. This implementation of OpenGL is based on OpenGL ES 3.0 but its architecture is more friendly to low end/mobile devices.
+
+While there are a few loose ends, contributor David Snopek has added stereoscopic rendering support to the OpenGL renderer and it has proven to be a very efficient solution for XR. This is an upgrade for anyone using the GLES2 renderer in Godot 3 to consider. This implementation takes advantage of the improvements in OpenGL ES 3 but does not make use of the techniques the Godot 3 GLES3 renderer applied that were detrimental to performance on mobile XR. The only footnote is that the implementation requires the OpenGL multiview extension.
+
+#### WebXR support in Godot 4
+
+The added OpenGL support has enabled us to port our WebXR implementation to Godot 4 as well, again thanks to David Snopek.
+
+Great care was taken to make WebXR support as portable as possible allowing games developed with Godot to be deployed to both OpenXR and WebXR based platforms.
+
+The only footnote is that WebXR has fixed actions bound to controller inputs which puts restrictions on the setup of actions on the OpenXR side to ensure portability between the platforms.
+
+For a future version of Godot 4 we are planning to add WebXR support to the action map implementation as well.
+
+#### Support for PICO headsets in Godot 4
+
+Support for PICO headsets was recently added to Godot 4, mostly to the credit of Dirk Steinmetz with help from several people at Bytedance. While we're still waiting for a driver update in order to fully support Vulkan, the OpenGL renderer is working fine.
+
+We hope to soon share instructions for setting up Godot on your PICO headset.
+
+#### Support for Magic Leap 2 headset in Godot 4
+
+Support for the Magic Leap 2 is in its final stages of development as well, the credit here goes to Ron Bessems.
+
+This is the first HMD we are supporting through the official Khronos loader and in theory this should provide support for several other headsets as well.
+
+Again we hope to soon share instructions for setting up Godot on your Magic Leap 2 headset.
+
+#### Honerary mention
+
+Godot 4 is getting a DirectX backend, this backend doesn't yet have support for stereoscopic rendering but adding this is on the roadmap and with that we're planning to add DirectX support to our OpenXR implementation as well. This will mean native WMR support for PCVR.
```

Comment:
It's postponed to 4.1: https://github.com/godotengine/godot/pull/70315#issuecomment-1403613525

```suggestion
Godot 4.1 will get a Direct3D 12 backend. This backend doesn't yet have support for stereoscopic rendering, but adding this is on the roadmap and with that we're planning to add Direct3D support to our OpenXR implementation as well. This will mean having native Windows Mixed Reality support for PCVR.
```